fluid/PaddleCV/human_pose_estimation/val.py

```diff
@@ -0,0 +1,321 @@
+add_arg('total_images',     int,   144406,              "Training image number.")
+add_arg('kp_dim',           int,   16,                  "Class number.")
+add_arg('model_save_dir',   str,   "output",            "Model save directory")
+add_arg('with_mem_opt',     bool,  False,               "Whether to use memory optimization or not.")
+add_arg('pretrained_model', str,   None,                "Whether to use pretrained model.")
+add_arg('checkpoint',       str,   None,                "Whether to resume checkpoint.")
+add_arg('lr',               float, 0.001,               "Set learning rate.")
+add_arg('lr_strategy',      str,   "piecewise_decay",   "Set the learning rate decay strategy.")
+add_arg('flip_test',        bool,  True,                "Flip test")
+add_arg('shift_heatmap',    bool,  True,                "Shift heatmap")
+add_arg('post_process',     bool,  True,               "Post process")
+# yapf: enable
+
+def valid(args):
+    if args.dataset == 'coco':
+        import lib.coco_reader as reader
+        IMAGE_SIZE = [288, 384]
+        HEATMAP_SIZE = [72, 96]
+        FLIP_PAIRS = [[1, 2], [3, 4], [5, 6], [7, 8], [9, 10], [11, 12], [13, 14], [15, 16]]
+        args.kp_dim = 17
+        args.total_images = 144406 # 149813
+    elif args.dataset == 'mpii':
+        import lib.mpii_reader as reader
+        IMAGE_SIZE = [384, 384]
+        HEATMAP_SIZE = [96, 96]
+        FLIP_PAIRS = [[0, 5], [1, 4], [2, 3], [10, 15], [11, 14], [12, 13]]
+        args.kp_dim = 16
+        args.total_images = 2958 # validation
+    else:
+        raise ValueError('The dataset {} is not supported yet.'.format(args.dataset))
+
+    print_arguments(args)
+
+    # image and target
+    image = layers.data(name='image', shape=[3, IMAGE_SIZE[1], IMAGE_SIZE[0]], dtype='float32')
+    target = layers.data(name='target', shape=[args.kp_dim, HEATMAP_SIZE[1], HEATMAP_SIZE[0]], dtype='float32')
+    target_weight = layers.data(name='target_weight', shape=[args.kp_dim, 1], dtype='float32')
+    center = layers.data(name='center', shape=[2,], dtype='float32')
+    scale = layers.data(name='scale', shape=[2,], dtype='float32')
+    score = layers.data(name='score', shape=[1,], dtype='float32')
+
+    # build model
+    model = pose_resnet.ResNet(layers=50, kps_num=args.kp_dim)
+
+    # output
+    loss, output = model.net(input=image, target=target, target_weight=target_weight)
+    loss.persistable = True
+    output.persistable = True
+    target.persistable = True
+
+    # parameters from model and arguments
+    params = {}
+    params["total_images"] = args.total_images
+    params["lr"] = args.lr
+    params["num_epochs"] = args.num_epochs
+    params["learning_strategy"] = {}
+    params["learning_strategy"]["batch_size"] = args.batch_size
+    params["learning_strategy"]["name"] = args.lr_strategy
+
+    if args.with_mem_opt:
+        fluid.memory_optimize(fluid.default_main_program())
```

Comment:
Betther to use [`skip_opt_set`](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/transpiler/memory_optimization_transpiler.py#L457) to skip the `loss`, `output` and `target`, instead of  setting

```
    loss.persistable = True
    output.persistable = True
    target.persistable = True
```


Same as other files.

Reply:
Done.